tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -15690,7 +15690,7 @@ void TryAddShaderAttrFromTargetProfile(Sema &S, FunctionDecl *FD,
 
   // if this FD isn't the entry point, then we shouldn't add
   // a shader attribute to this decl, so just return
-  if (EntryPointName != FD->getIdentifier()->getName()) {
+  if (FD->getIdentifier() && EntryPointName != FD->getIdentifier()->getName()) {
```

Comment:
I don't think this is right.  We should return if there is no identifier as well, since it can't be an entry point.

```suggestion
    if (!FD->getIdentifier() || EntryPointName != FD->getIdentifier()->getName()) {
```